types/algoliasearch.d.ts

```diff
@@ -0,0 +1,148 @@
+// Note: Below, we will be importing all algoliasearch v3,4,5 types.
+// The goal is being able to export the algoliasearch-helper types using
+// the version of the client installed by the developer.
+
+// @ts-ignore
+import type algoliasearch from 'algoliasearch/lite';
+// @ts-ignore
+import type * as AlgoliaSearchLite from 'algoliasearch/lite';
+// @ts-ignore
+import type * as AlgoliaSearch from 'algoliasearch';
+// @ts-ignore
+import type * as ClientSearch from '@algolia/client-search';
+
+// turns any to unknown, so it can be used as a conditional
+type AnyToUnknown<T> = (any extends T ? true : false) extends true
+  ? unknown
+  : T;
+
+type DummySearchClientV4 = {
+  transporter: unknown;
+};
+
+type DummySearchClient = {
+  search: unknown;
+};
```

Comment:
Are these `Dummy*` types already being used and accessible to users? The term is ableist so if we have the ability to rename it, we should.

Reply:
I'd not heard it in the insulting context before (only as a model). They aren't exposed so we can rename to whatever we want. is MockSearchClient or SearchClientShape fine?

Reply:
Yeah it's listed in the ableist terms to avoid of the Algolia docs (inherited from the [Google styleguide](https://developers.google.com/style/inclusive-documentation#ableist-language)), and many other sources flag it as a problematic term.

`MockSearchClient` sounds fine, especially if that's a type we could reuse for tests, that'd make sense to me.